### PR TITLE
new method submitApplication

### PR DESCRIPTION
**Pull request by balcirakpeter**
- new method submitApplication was created, which works same as
  createApplication but it returns application instead of an application
  id. It will be used by perun-wui.
- private method processApplication was created to prevent duplicities.
  It returns application id. According the id will be retrieved data or
  application.

### CODE REVIEW

File: perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/RegistrarManager.java [hunk 1]
```diff
@@ -229,6 +229,19 @@
...
+	 *
+	 * <p>The method triggers approval for VOs with auto-approved applications.
+	 *
+	 * @param session user present in session
+	 * @param application application
+	 * @param data data
+	 * @return submitted application
+	 * @throws PerunException
+	 */
+	Application submitApplication(PerunSession session, Application application, List<ApplicationFormItemData> data) throws PerunException;
```
Comment (zlamalp) [inline perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/RegistrarManager.java:243]: We probably can mark old ``createApplication()`` as deprecated (in core and RPC javadoc). I don't think anybody else is using it directly.

File: perun-rpc/src/main/java/cz/metacentrum/perun/rpc/methods/RegistrarManagerMethod.java [hunk 2]
```diff
@@ -601,6 +601,29 @@ public Application call(ApiCaller ac, Deserializer parms) throws PerunException
 
 	},
 
+	/*#
+	 * Creates a new application.
+	 * The method triggers approval for VOs with auto-approved applications.
+	 *
+	 * @param app Application Application JSON object
+	 * @param data List<ApplicationFormItemData> List of ApplicationFormItemData JSON objects
+	 * @return submitted Application
```
Comment (zlamalp) [inline perun-rpc/src/main/java/cz/metacentrum/perun/rpc/methods/RegistrarManagerMethod.java:610]: You must specify data type in return for RPC docs: ``@return Application Submitted application``

File: perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/RegistrarManager.java (hunk 1 above)
Comment (balcirakpeter) [reply perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/RegistrarManager.java:244]: Done

File: perun-rpc/src/main/java/cz/metacentrum/perun/rpc/methods/RegistrarManagerMethod.java (hunk 2 above)
Comment (balcirakpeter) [reply perun-rpc/src/main/java/cz/metacentrum/perun/rpc/methods/RegistrarManagerMethod.java]: Fixed

**Comment by balcirakpeter** [comment]
> Please, add this new method to the OpenAPI specification.

Done

File: perun-rpc/src/main/java/cz/metacentrum/perun/rpc/methods/RegistrarManagerMethod.java [hunk 3]
```diff
@@ -582,6 +582,7 @@ public Application call(ApiCaller ac, Deserializer parms) throws PerunException
 	 * Creates a new application.
 	 * The method triggers approval for VOs with auto-approved applications.
 	 *
+	 * @deprecated
```
Comment (zlamalp) [inline perun-rpc/src/main/java/cz/metacentrum/perun/rpc/methods/RegistrarManagerMethod.java:585]: Let docs reader know about the new way: ``@deprecated See submitApplication()``

Comment (balcirakpeter) [reply perun-rpc/src/main/java/cz/metacentrum/perun/rpc/methods/RegistrarManagerMethod.java]: Done

File: perun-openapi/openapi.yml [hunk 4]
```diff
@@ -11263,6 +11263,33 @@ paths:
...
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              title: InputSubmitApplication
+              description: "input to submitApplication"
+              type: object
+              required:
+                - application
```
Comment (Vojtech-Sassmann) [inline perun-openapi/openapi.yml:11287]:
```suggestion
                - app
```

File: perun-openapi/openapi.yml [hunk 5]
```diff
@@ -11263,6 +11263,33 @@ paths:
...
+                - data
+              properties:
+                application: { $ref: '#/components/schemas/Application' }
```
Comment (Vojtech-Sassmann) [inline perun-openapi/openapi.yml:11290]:
```suggestion
                app: { $ref: '#/components/schemas/Application' }
```

File: perun-openapi/openapi.yml (hunk 4 above)
Comment (balcirakpeter) [reply perun-openapi/openapi.yml]: Fixed

File: perun-openapi/openapi.yml (hunk 5 above)
Comment (balcirakpeter) [reply perun-openapi/openapi.yml]: Fixed

**Pull request merged**